lib/logger/lib/logger.ex

```diff
@@ -130,6 +130,13 @@ defmodule Logger do
       in the queue is reduced to `sync_threshold * 0.75` messages.
       Defaults to 20 messages.
 
+    * `:discard_threshold` - if the `Logger` manager has more than
+      `:discard_threshold` messages in its queue, `Logger` will change
+      to *discard mode* and messages will be discarded directly in the
+      clients. `Logger` will return to *sync mode* once the number of
+      messages in the queue is reduced to `discard_threshold * 0.75`
+      messages. Defaults to 10000 messages.
```

Comment:
10000 is a very big number, there could be a lot of reference counted binaries in there or other large data structures using up memory. The GC time on these processes will be significant as we have on heap message queues - we could switch to off heap for Logger.

Also the time to process 7500 messages is still likely to be a long time to be stuck in sync mode (which has infinity timeout!). If the sync mode is 20 and the discard is 10000 chances are your app is already broken and remains broken regardless because the sync mode did no provide back pressure to the application, so likely the logs are coming in from non-backpressure requests (e.g. the node is a server and logging based on external requests) and those are going to be very slow. I think we should default to a number closer to 500 like before.

Another issue with a big number is we can end up with "buffer bloat" and are more likely to have empty periods where there are no logs.

Reply:
What do you suggest? 1000? 2000? 5000?

Reply:
Staying with 500 is ok I think as we had that before.